main.go

```diff
@@ -157,18 +157,30 @@ func main() {
 		"PullRequest":             generators.NewPullRequestGenerator(mgr.GetClient()),
 	}
 
-	combineGenerators := map[string]generators.Generator{
-		"Matrix": generators.NewMatrixGenerator(baseGenerators),
+	nestedGenerators := map[string]generators.Generator{
+		"List":                    terminalGenerators["List"],
+		"Clusters":                terminalGenerators["Clusters"],
+		"Git":                     terminalGenerators["Git"],
+		"SCMProvider":             terminalGenerators["SCMProvider"],
+		"ClusterDecisionResource": terminalGenerators["ClusterDecisionResource"],
+		"PullRequest":             terminalGenerators["PullRequest"],
+		"Matrix":                  generators.NewMatrixGenerator(terminalGenerators),
+		"Union":                   generators.NewUnionGenerator(terminalGenerators),
 	}
 
-	all, err := generators.CombineMaps(baseGenerators, combineGenerators)
-	if err != nil {
-		setupLog.Error(err, "generators can't be combined")
-		os.Exit(1)
+	topLevelGenerators := map[string]generators.Generator{
+		"List":                    terminalGenerators["List"],
+		"Clusters":                terminalGenerators["Clusters"],
+		"Git":                     terminalGenerators["Git"],
+		"SCMProvider":             terminalGenerators["SCMProvider"],
+		"ClusterDecisionResource": terminalGenerators["ClusterDecisionResource"],
+		"PullRequest":             terminalGenerators["PullRequest"],
+		"Matrix":                  generators.NewMatrixGenerator(nestedGenerators),
+		"Union":                   generators.NewUnionGenerator(nestedGenerators),
```

Comment:
These annoy me because of the maintenance overhead (when a new generator is added, you have to remember to add it in all three places). But I can't think of a better alternative.